temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java

```diff
@@ -1385,6 +1386,20 @@ public static String getCurrentDetails() {
     return WorkflowInternal.getCurrentDetails();
   }
 
+  /**
+   * Get the currently running workflow instance.
+   *
+   * @apiNote The instance is only available after it has been initialized. This function will
+   *     return null if called before the workflow has been initialized. For example, this could
+   *     happen if the function is called from a {@link WorkflowInit} constructor or {@link
+   *     io.temporal.common.interceptors.WorkflowInboundCallsInterceptor#init(WorkflowOutboundCallsInterceptor)}.
```

Comment:
So long as available in `execute` of this same interceptor, I think we're good

Reply:
Yeah in the test I show this is not null in `execute`